packages/a11y-base/src/field-aria-controller.js

```diff
@@ -37,7 +37,9 @@ export class FieldAriaController {
   setTarget(target) {
     this.__target = target;
     this.__setAriaRequiredAttribute(this.__required);
-    this.__setLabelIdToAriaAttribute(this.__labelId);
+    const isFromUser = this.__labelIdFromUser != null;
```

Comment:
**idea:** When reading the code, I got the impression that it might be more clear if the controller provided an individual method for setting `accessible-ref` rather than supporting two branches inside `setLabelId`. I kind of understand that the idea was likely to keep the controller abstract but that doesn't seem to help much for a new reader like me. The same probably goes for `accessible-name`.

- `setAccessibleRefId()`
- `setAccessibleName()`

Having the naming aligned with the component's properties can reduce cognitive load.

Example:

```js
this.__setLabelIdToAriaAttribute(this.__labelId);
this.__setAccessibleRefIdToAriaAttribute(this.__accessibleRefId);
this.__setAccessibleNameToAriaAttribute(this.__accessibleName);
```

This idea isn't directly related to the current PR.

Reply:
**question:** By the way, what ensures that the label id is reverted when the accessible ref id is removed?

Reply:
> **question:** By the way, could you point out what reverts the label node id when the accessible ref id is removed?

This is how the `setAriaIDReference(target, attr, config = { newId: null, oldId: null, fromUser: false })` in `aria-id-reference.js` works:
- after you call the method with the `fromUser` flag on in the configuration object and some value to `newId`, it stores the current value in `attr` and replaces it with `newId`
- if in a later call, with `fromUser` on and `newId == null`, the function restores the stored value

The idea is to have a way to define when the attribute is being controlled by the user (`fromUser` on and some value in `newId`) and when the user "gives back" the control to the component (`fromUser` on with falsy value in `newId`).